### PR TITLE
Add documentation for Layout/MultilineAssignmentLayout `SupportedTypes`

### PR DESCRIPTION
**Pull request by Bhacaz**
I was looking for exactly the configurable attributes `SupportedTypes` for the cop Layout/MultilineAssignmentLayout.  But, I had to look in the code to found out this attribute and in `config/default.yml` to found the available parameters. I thought it might be useful to add this info in the documentation. 

UPDATE

To make `SupportedTypes` appear in the documentation, I had to change the name of the attribute, I named it `AssignmentTypes`.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop-hq/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by bbatsov** [comment]
The documentation is auto-generated from the cop source, so you'll actually have to update it instead.

**Comment by Bhacaz** [comment]
@bbatsov 

> The documentation is auto-generated from the cop source, so you'll actually have to update it instead.

Sorry I totally missed that. But now I understand why the attribute wasn’t present in the documentation.

--- 

The reason `SupportedStyles` wasn't appearing in the documentation is because of this line: https://github.com/rubocop-hq/rubocop/blob/bfb6a0607a19531ac8e1f20fe27a30a9da29dede/lib/rubocop/cops_documentation_generator.rb#L120

I don't know if was on purpose to name it that way (I don't think so), but I think it will be interesting to add documentation about that config attribute.

I had two choice that imply to change the config name used by the cop:
1. Use the pattern `EnforcedTypes`/`SupportedTypes`. Which required to specified every possible supported values and use  `EnforcedTypes` in the code instead of `SupportedTypes`.
2. Change the config name, `AssignmentTypes` and the docs will only specified that it required an `Array`.

**Comment by bbatsov** [comment]
> I don't know if was on purpose to named it that way (I don't think so), but I think it will be interesting to add documentation about that config attribute.

@koic Do you remember why we did it like this? I don't want to change the attribute name just to fix the docs, as that'd be a breaking change.

**Comment by koic** [comment]
I agree with @bbatsov. I also think that renaming option is a breaking change, so it's better to only update the documentation for `SupportedTypes` now.

JFYI, It seems that the option has been named `SupportedTypes` for a long time.
https://github.com/rubocop-hq/rubocop/pull/2361

**Comment by Bhacaz** [comment]
@koic 
I also think that a breaking change is too must to fix the documentation. What I can see, is to have a complet documentation for that attribute it might required that. The table of _Configurable attribute_ looks for the pattern `Supported*/Enforced*` names and to configure that attribute in a `.rubocop.yml` file we must used `EnforcedTypes` (instead of the actual `SupportedStyles`).

I have a suggestion, I can only add example in the cop code for that attribute and it will be shown in the documentation, but `SupportedTypes` will be missed in the table of _Configurable attribute_.

<details>

image

</details>

Let me know what you think.

**Comment by koic** [comment]
I see. It's not perfect as configuration table, but I think it can be merged as it  is as a first step (I think that examples has been improved by this patch). And I think RuboCop 2.0 is an opportunity for breaking change to the better option name.

File: changelog/change_add_supported_types_documentation_multiline_assignment.md [hunk 1]
```diff
@@ -0,0 +1 @@
+* [#9397](https://github.com/rubocop-hq/rubocop/issues/9397): Update Layout/MultilineAssignmentLayout examples to add `SupportedTypes` attribute in documentation. ([@bhacaz][])
```
Comment (koic) [inline changelog/change_add_supported_types_documentation_multiline_assignment.md:1]: Can you remove the entry as there is no change in product behavior to users?

**Comment by koic** [comment]
Thanks!

**Pull request merged**